mantaray/received.py

```diff
@@ -463,13 +507,21 @@ def _handle_received_message(
         assert isinstance(msg, backend.MessageFromUser)
         _handle_kick(server_view, msg.sender_nick, msg.args)
 
+    elif msg.command == "AWAY":
+        assert msg.sender is not None
+        _handle_away(server_view, msg.sender, msg.args)
+
     elif msg.command == "CAP":
         _handle_cap(server_view, msg.args)
 
     elif msg.command == "AUTHENTICATE":
         _handle_authenticate(server_view)
 
-    elif msg.command == RPL_LOGGEDIN:
+    elif msg.command == RPL_WELCOME:
+        server_view.core.cap_req = []
+        server_view.core.cap_list = set()
+
+    elif msg.command == RPL_SASLSUCCESS or msg.command == ERR_SASLFAIL:
         server_view.core.send("CAP END")
```

Comment:
We should probably add the received text into the server view. Currently this doesn't display an error message anywhere when it receives `ERR_SASLFAIL`.